app/page/template/content/preferences-devices.htm

```diff
@@ -5,13 +5,17 @@
       <section class="preferences-section" data-uie-name="preferences-device-current">
         <header class="preferences-header" data-bind="l10n_text: z.string.preferences_devices_current"></header>
         <div class="preferences-devices-model" data-bind="text: current_client().model"></div>
-        <div class="preferences-devices-id"><span data-bind="l10n_text: z.string.preferences_devices_id"></span><span data-bind="html: z.util.print_devices_id(current_client().id)" data-uie-name="preferences-device-current-id"></span></div>
+        <div class="preferences-devices-id"><span data-bind="l10n_text: z.string.preferences_devices_id"></span>
+          <span data-bind="foreach: displayClientId().match(/.{1,2}/g)" data-uie-name="preferences-device-current-id">
+            <span class="device-id-part" data-bind="text: $data"></span>
+          </span>
+        </div>
         <div class="preferences-devices-activated">
-          <div data-bind="html: activated_on"></div>
-          <div data-bind="html: activated_in"></div>
+          <div data-bind="foreach: activated_on"><span class="preferences-devices-activated-bold" data-bind="text: $data">?</span></div>
+          <div data-bind="foreach: activated_in"><span class="preferences-devices-activated-bold" data-bind="text: $data">?</span></div>
         </div>
         <div class="preferences-devices-fingerprint-label" data-bind="l10n_text: z.string.preferences_devices_fingerprint"></div>
-        <div class="preferences-devices-fingerprint" data-bind="html: z.util.print_devices_id(fingerprint())"></div>
+        <div class="preferences-devices-fingerprint" data-bind="foreach: displayFingerPrint().match(/.{1,2}/g)"><span class="device-id-part" data-bind="text: $data"></span></div>
```

Comment:
Shouldn't the display fingerprint already contain the array instead of having additional logic in the HTML?